contracts/src/v0.8/vrf/dev/VRFV2PlusWrapper.sol

```diff
@@ -9,14 +9,20 @@ import {AggregatorV3Interface} from "../../shared/interfaces/AggregatorV3Interfa
 import {VRFV2PlusClient} from "./libraries/VRFV2PlusClient.sol";
 import {IVRFV2PlusWrapper} from "./interfaces/IVRFV2PlusWrapper.sol";
 import {VRFV2PlusWrapperConsumerBase} from "./VRFV2PlusWrapperConsumerBase.sol";
-import {ChainSpecificUtil} from "../../ChainSpecificUtil.sol";
+import {VRFV2PlusWrapperL1Fees} from "./VRFV2PlusWrapperL1Fees.sol";
 
 /**
  * @notice A wrapper for VRFCoordinatorV2 that provides an interface better suited to one-off
  * @notice requests for randomness.
  */
 // solhint-disable-next-line max-states-count
-contract VRFV2PlusWrapper is ConfirmedOwner, TypeAndVersionInterface, VRFConsumerBaseV2Plus, IVRFV2PlusWrapper {
+contract VRFV2PlusWrapper is
+  ConfirmedOwner,
+  TypeAndVersionInterface,
+  VRFConsumerBaseV2Plus,
+  IVRFV2PlusWrapper,
+  VRFV2PlusWrapperL1Fees
```

Comment:
@leeyikjiun What is the main reason for the introduction of this base contract: VRFV2PlusWrapperL1Fees?

Reply:
I added this contract. It's used to abstract away the L1 fee calculation for all L2 chains we support (this is something that the `ChainSpecificUtil` library did before) and also to use the same L1 fee calculation currently provided to the VRFCoordinator. `VRFV2PlusWrapperL1Fees` is just combining `OptimismL1Fees` and `ArbitrumL1Fees` contracts that are meant to extend the base `VRFCoordinatorV2_5` contract with L2-specific functionality.

Reply:
I agree with the new pattern generally. Only thing that is interesting now is that on the Arbitrum wrapper side, we can technically invoke `setL1FeeCalculation`, which is only meant for Optimism right? should we follow the same pattern as coordinator where we have `VRFV2PlusWrapperOptimism` and `VRFV2PlusWrapperArbitrum` as separate contracts?

Reply:
The reason why I took a different path for the wrapper is that I wanted to keep the number of diverged contracts to a minimum. The wrapper doesn't suffer from the contract size limit issues so we have more freedom to combine logic for different chains in the same contract.

I see the reason for concern with the invocation of the`setL1FeeCalculation` function on Arbitrum. I can suggest that this function should revert if someone tries to use it on Arbitrum or any other L2 chain not part of the OP stack. Of course, that still doesn't solve the issue of having this function available on Arbitrum in the first place.

On the other hand, I don't mind the option of diverging the wrapper contract to keep the same pattern and segregate chain-specific functionality into different contracts. This solution is more clean, but it does complicate things for our tooling. I think it's worth discussing.

Reply:
I don't have a strong opinion on this, either is fine for me. Perhaps, one reason to split to chain specific contracts is to avoid the sload when the chain doesn't have to. Splitting it up isn't too troublesome to maintain as well as we are unlikely to touch the L1Fees portion, so it's a create one new contract per L2 chain and be done with it thing.

On the tooling end, if there's a consistent pattern to it, then I _think_ it should be fine as well.

Reply:
Yeah, even I don't have a strong opinion on this. If splitting up the contracts reduces an sload in terms of gas and reduces contract size overhead, we could proceed to do that for the wrapper as well.
As @leeyikjiun mentions, we'll have a consistent pattern of different contracts for both VRFCoordinatorV2_5 and VRFV2PlusWrapper.
Also, the maintenance overhead of diverged contracts would not be a lot, since we're inheriting from the base VRFCoordinatorV2_5 and VRFV2PlusWrapper contracts. Most of the new changes would go to the base contracts with the L1Fees portion not changing much.

Reply:
I just want to let you know that I've done the same thing with the wrapper as it was done for the coordinator. That means now we have 3 wrapper contracts: one for Arbitrum, one for Optimism and one for L1 chains.